### PR TITLE
Add configuration header

### PR DESCRIPTION
**Pull request by azubieta**
The configuration header provides information of which features where enabled at build time and about the library version.

### CODE REVIEW

File: include/appimage/appimage++.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /**
  * C++ headers aggregation to differentiate from the C only interface.
  */
-
+#include <appimage/config.h>
```
Comment (TheAssassin) [inline include/appimage/appimage++.h]: Empty line missing.

File: src/libappimage/CMakeLists.txt [hunk 2]
```diff
@@ -70,6 +76,12 @@ install(
     COMPONENT libappimage-dev
 )
 
+install(
```
Comment (TheAssassin) [inline src/libappimage/CMakeLists.txt]: I think we should move the generated headers in some directory with a different, more meaningful name, e.g., `generated-headers/appimage`. `appimage` as a name is ambiguous.

File: src/libappimage/config.h.in [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// LibAppImage Version
+
+#define LIBAPPIMAGE_VERSION_MAJOR @V_MAJOR@
+#define LIBAPPIMAGE_VERSION_MINOR @V_MINOR@
+#define LIBAPPIMAGE_VERSION_PATCH @V_PATCH@
+#define LIBAPPIMAGE_VERSION_SUFFIX @V_SUFFIX@
+
+#define LIBAPPIMAGE_VERSION_STR "@libappimage_VERSION@"
```
Comment (TheAssassin) [inline src/libappimage/config.h.in]: Why the `_STR` suffix? Should match what's defined in CMake, i.e., `LIBAPPIMAGE_VERSION`.

Also, do we set either of those manually in CMake somewhere? If yes, it should be removed there to make sure the config header works.

Comment (azubieta) [reply src/libappimage/config.h.in]: > Why the _STR suffix?

Processor defines doesn't have a type, this suffix gives a hint about it.

Comment (TheAssassin) [reply src/libappimage/config.h.in]: I don't think that's really necessary, people use such vars as documented... I don't like the inconsistency. You can also rename the CMake variable.

Comment (azubieta) [reply src/libappimage/config.h.in]: > Also, do we set either of those manually in CMake somewhere? If yes, it should be removed there to make sure the config header works.

It's used mostly in the test. But yes it should be properly tested, my bad.

File: include/appimage/appimage++.h (hunk 1 above)
Comment (azubieta) [reply include/appimage/appimage++.h]: fixed

File: src/libappimage/CMakeLists.txt (hunk 2 above)
Comment (azubieta) [reply src/libappimage/CMakeLists.txt]: done

File: src/libappimage/config.h.in (hunk 3 above)
Comment (azubieta) [reply src/libappimage/config.h.in]: > people use such vars as documented...

They are not documented. It's a common practice using the `_STR` prefix but it's also true the opposite. If you prefer it without the suffix let's change it.

Comment (TheAssassin) [reply src/libappimage/config.h.in]: I don't care really. I do see the purpose of the suffix. Therefore the suggestion to just add it in CMake too.

Comment (azubieta) [reply src/libappimage/config.h.in]: Too late. Changing the var name in CMake will require to check/test all it's usages. Let's not touch what is already working.

**Comment by TheAssassin** [comment]
Have you removed the definitions of the variables in CMake already?

**Comment by azubieta** [comment]
Yes, look at:

https://github.com/AppImage/libappimage/pull/124/files#diff-db693dd81c3b83d0754285a5a0b86731L258

https://github.com/AppImage/libappimage/pull/124/files#diff-db693dd81c3b83d0754285a5a0b86731L262

**Comment by TheAssassin** [comment]
Ah, okay, I thought there was more of that. Thanks.

**Pull request merged**